pkg/util/ecs/ecs.go

```diff
@@ -29,6 +29,9 @@ const (
 	DefaultAgentPort = 51678
 	// Cache the fact we're running on ECS Fargate
 	isFargateInstanceCacheKey = "IsFargateInstanceCacheKey"
+
+	// CloudProviderName contains the inventory name of for ECS
+	CloudProviderName = "AWS"
```

Comment:
this will create duplicate entries (on key "AWS") in your map of cloud provider detectors, which will make the Agent only use the ECS metadata collector or the EC2 metadata collector to detect if it's on AWS.

Instead, we need to add some logic to report that we're running on AWS if we detect any of: ECS, EC2, etc

Also, it could make more sense to define the different `CloudProviderName`s directly in `cloudprovider.go`, since an important property is to ensure the uniqueness of these names, and we can ensure that statically in the map of `cloudprovider.go`.

Reply:
Good catch. I think it's important that the logic for each cloudprovider stays in the cloud provider package. I updated the code